### PR TITLE
Implement PSR-11

### LINKED ISSUE #13052: [DIC] Implement PSR-11
The [PSR-11](https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-11-container.md) standardizes the way entries are retrieved from a container.

`PMA\libraries\di\Container` should implement `Psr\Container\ContainerInterface`.

- https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-11-container.md
- https://packagist.org/packages/psr/container

### PR DESCRIPTION
**Pull request by MauricioFauth**
Fixes #13052

Signed-off-by: Maurício Meneghini Fauth <mauriciofauth@gmail.com>

Before submitting pull request, please check that every commit:

- [x] Has proper Signed-Off-By
- [x] Has commit message which describes it
- [ ] Is needed on it's own, if you have just minor fixes to previous commits, you can squash them
- [ ] Any new functionality is covered by tests

### CODE REVIEW

**Comment by mauriciofauth** [comment]
I've never written tests before, so I do not know if the ones I've written are correct.

File: libraries/di/Container.php [hunk 1]
```diff
@@ -46,19 +47,58 @@ public function __construct(Container $base = null)
...
     public function get($name, $params = array())
     {
+        try {
+            if (!$this->has($name)) {
+                throw new NotFoundException("No entry was found for $name identifier.");
+            }
+        } catch (NotFoundException $e) {
+            echo $e->getMessage();
+            return null;
+        }
```
Comment (nijel) [inline libraries/di/Container.php]: I really don't get the code here:

- Why is there `try { throw } catch {}` block at all?
- The echo is there for something else than debugging purposes?

File: libraries/di/Container.php [hunk 2]
```diff
@@ -46,19 +47,58 @@ public function __construct(Container $base = null)
      * @param string $name   Name
      * @param array  $params Parameters
      *
+     * @throws NotFoundException  No entry was found for **this** identifier.
+     * @throws ContainerException Error while retrieving the entry.
```
Comment (nijel) [inline libraries/di/Container.php]: It doesn't seem to throw these as they are always handled inside...

File: libraries/di/Container.php [hunk 3]
```diff
@@ -46,19 +47,58 @@ public function __construct(Container $base = null)
...
+            return true;
         }
 
         if (isset($GLOBALS[$name])) {
-            return $GLOBALS[$name];
+            return true;
         }
 
-        return null;
+        return false;
```
Comment (nijel) [inline libraries/di/Container.php]: Probably better to write whole method as:

```php
    return isset($this->content[$name]) || isset($GLOBALS[$name]);
```

File: test/classes/di/ContainerTest.php [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    }
+
+    /**
+     * Test for get
+     * @expectedExceptionMessageRegExp #Right.*#
+     * @return void
+     */
+    public function testGetWithInvalidEntry()
+    {
+        $this->assertSame(null, $this->container->get('invalid'));
```
Comment (nijel) [inline test/classes/di/ContainerTest.php]: IMHO it should throw NotFoundExceptionInterface according to PSR-11.

**Review by nijel — CHANGES_REQUESTED** [review]

**Comment by nijel** [comment]
And now the code code which uses container needs to be adjusted to understand new behavior, see at least the test failures: https://travis-ci.org/phpmyadmin/phpmyadmin/jobs/210553712#L457

**Comment by mauriciofauth** [comment]
I will do this @nijel 

And about the `get` method, I think it would not be appropriate to use `isset` since this already happens in the `has` method. I think it should check whether the value is valid or not. Maybe `!is_null`?

**Comment by nijel** [comment]
Why should it check for null value? I think it's perfectly okay to return null if it has been set...

**Comment by mauriciofauth** [comment]
Oh, of course. I was not thinking straight.
Looking at the code, I realized that all values are instances of `ValueItem`. Maybe check for it.
I'm wanting to check for something other than `isset`, otherwise `ContainerException` will never be thrown.

**Comment by mauriciofauth** [comment]
I left `get` method the way it was and corrected the tests that were failing.
Is everything all right now, @nijel?

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Pull request merged**